plugins/out_s3/s3.c

```diff
@@ -118,6 +118,45 @@ static char *mock_error_response(char *error_env_var)
     return NULL;
 }
 
+static void s3_retry_warn(struct flb_s3 *ctx, const char *tag,
+                          char *input_name, time_t create_time,
+                          int less_than_limit)
+{
+    struct tm now_time;
+    char create_time_str[20];
+    struct tm *tmp;
+
+    tmp = localtime_r(&create_time, &now_time);
+    strftime(create_time_str, 20, "%Y-%m-%d %H:%M:%S", tmp);
+    if (input_name == NULL) {
+        if (less_than_limit == FLB_TRUE) {
+            flb_plg_warn(ctx->ins,
+                        "failed to flush chunk tag=%s, create_time=%s, "
+                        "retry issued: (out_id=%d)",
+                        tag, create_time_str, ctx->ins->id);
+        }
+        else {
+            flb_plg_warn(ctx->ins,
+                        "chunk tag=%s, create_time=%s cannot be retried",
+                        tag, create_time_str);
+        }   
+    }
+    else if (strlen(input_name) > 0) {
```

Comment:
In the future it might have been better to put this in the upper check as:
input_name == NULL || strlen(input_name) == 0

That way there isn't a case that is not covered, specifically:
strlen(input_name) == 0
where input_name is an empty string ""

Reply:
I think we can leave it as is for now.

Reply:
I'm confused- `strlen("")` is 0?